api/v1alpha1/etcd_types.go

```diff
@@ -242,6 +242,20 @@ type SharedConfig struct {
 	AutoCompactionRetention *string `json:"autoCompactionRetention,omitempty"`
 }
 
+// SchedulingConstraints defines the different scheduling constraints that must be applied to the
+// pod spec in the etcd statefulset.
+// Currently supported constraints are Affinity and TopologySpreadConstraints.
+type SchedulingConstraints struct {
+	// Affinity defines the various affinity and anti-affinity rules for a pod
+	// that are honoured by the kube-scheduler.
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// TopologySpreadConstraints describes how a group of pods ought to spread across topology domains,
+	// that are honoured by the kube-scheduler.
+	// +optional
+	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
```

Comment:
Minor point: do you still want to keep TSC? We decided that we will not be using TSC. When these are stable and later if we feel that these should be used, then we can add them as well.

Reply:
We are still unsure whether we want to use TSC or not. Maybe it still might make sense for our usecase since e won't need to scale beyond 5 replicas. With this PR, I have just provided the option of using these scheduling constraints, while the gardenlet can actually take the decision of whether to use TSC or affinity or maybe a combination of both. This PR allows us to keep our options open.

Reply:
Ok, this is fine.